lib/Serialization/Deserialization.cpp

```diff
@@ -3986,15 +3986,9 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
     break;
   }
 
-  case decls_block::LVALUE_TYPE: {
-    TypeID objectTypeID;
-    decls_block::LValueTypeLayout::readRecord(scratch, objectTypeID);
-    typeOrOffset = LValueType::get(getType(objectTypeID));
-    break;
-  }
   case decls_block::INOUT_TYPE: {
     TypeID objectTypeID;
-    decls_block::LValueTypeLayout::readRecord(scratch, objectTypeID);
+    decls_block::InOutTypeLayout::readRecord(scratch, objectTypeID);
```

Comment:
Before, we were serializing INOUT_TYPE using LValueTypeLayout. Is this what's causing the test failure here?